File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int n = numbers.size();
        int i=0;
        int j=n-1;
        // vector<int> ans;
        while(i<=j){
            int sum=0;
            sum = numbers[i]+numbers[j];
            if(sum==target){
                return{i+1,j+1};
            }
            else if(sum>target){
                j--;
            }
            else{
                i++;
            }
        }
        return {-1,-1};
    }
};

File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int start = 0;
        int end = numbers.length-1;
        int[] ans = new int[2];
        while(start<end){
            int sum = numbers[start] + numbers[end];
            if(sum<target){
                start++;
            }
            else if(sum>target){
                end--;
            }
            else{
                ans[0] = start+1;
                ans[1] = end+1;
                break;
            }
        }
        return ans;
    }
}